### PR TITLE
Feature/voip api

### PR DESCRIPTION
**Pull request by Jisay**
Beginning of reworking services and factories.
Use it on telephony sidebar section

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ module.exports = {
         "no-underscore-dangle": 0,
```
Comment (JeremyDec) [inline .eslintrc.js]: Attention, set develop branch in place of master

File: client/components/telecom/voip/service/voip-service.service.js [hunk 2]
```diff
@@ -32,10 +34,125 @@ angular.module("managerApp").service("voipService", class {
...
+     *  Use API to get single service of given billingAccount and serviceName
+     *
+     *  @param  {String} billingAccount The billingAccount to which is attached the service.
+     *  @param  {String} serviceName    The unique id of the service.
+     *
+     *  @return {Promise}   That returns a VoipService instance representing the service fetched.
+     */
+    fecthSingleService (billingAccount, serviceName) {
+        return this.ovhApiTelephony.Service().Lexi().get({
+            billingAccount: billingAccount,
```
Comment (antleblanc) [inline client/components/telecom/voip/service/voip-service.service.js]:
```js
{
    billingAccount,
    serviceName
}
```

File: client/components/telecom/telephony/sidebar/telephony-sidebar.service.js [hunk 3]
```diff
@@ -9,91 +9,114 @@ angular.module("managerApp").service("TelephonySidebar", function ($q, $translat
...
+            let sortedBillingAccounts = billingAccounts.sort(function (first, second) {
                 return first.getDisplayedName().localeCompare(second.getDisplayedName());
             });
-            _.forEach(sortedGroups, function (group) {
 
-                var subSection = SidebarMenu.addMenuItem({
-                    id: group.billingAccount,
-                    title: group.getDisplayedName(),
+            // add billingAccount subsections to telephony sidebar menu item
+            sortedBillingAccounts.forEach(function (billingAccount) {
```
Comment (JeremyDec) [inline client/components/telecom/telephony/sidebar/telephony-sidebar.service.js]: suggestion : maybe you could factorize forEach function in each of sorted items no ?

File: client/components/telecom/voip/service/voip-service.service.js (hunk 2 above)
Comment (antleblanc) [inline client/components/telecom/voip/service/voip-service.service.js]: You can use shorthand if you prefer :)
```js
{
    billingAccount,
    serviceName
}
```

File: client/components/telecom/voip/service/line/voip-service-line.factory.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        /**
+         *  @description
+         *  Helper that try to check if the service offer is trunk or not.
+         *
+         *  @return {Boolean}
+         */
+        isSipTrunk () {
+            var publicOfferName = _.get(this.getPublicOffer, "name");
```
Comment (antleblanc) [inline client/components/telecom/voip/service/line/voip-service-line.factory.js]: `var` => `const`.

File: client/components/telecom/telephony/sidebar/telephony-sidebar.service.js (hunk 3 above)
Comment (Jisay) [reply client/components/telecom/telephony/sidebar/telephony-sidebar.service.js]: ho yeah. I though to that and then I forgot :smile:

File: .eslintrc.js (hunk 1 above)
Comment (Jisay) [reply .eslintrc.js]: done

File: client/components/telecom/voip/service/voip-service.service.js [hunk 5]
```diff
@@ -51,7 +51,7 @@ angular.module("managerApp").service("voipService", class {
      *  @return {Promise}   That returns a VoipService instance representing the service fetched.
      */
     fecthSingleService (billingAccount, serviceName) {
```
Comment (JeremyDec) [inline client/components/telecom/voip/service/voip-service.service.js]: typo : fecthSingleService => fetchSingleService :)

File: client/components/telecom/telephony/sidebar/telephony-sidebar.service.js [hunk 6]
```diff
@@ -64,7 +64,7 @@ angular.module("managerApp").service("TelephonySidebar", function ($translate, S
 
                 // display lines except plugAndFax and fax
                 let sortedSipLines = _.filter(sortedLines, function (line) {
-                    return line.featureType !== "plugAndFax" && line.featureType !== "fax";
+                    return line.featureType !== "plugAndFax" && line.featureType !== "fax" && line.featureType !== "voicefax";
```
Comment (JeremyDec) [inline client/components/telecom/telephony/sidebar/telephony-sidebar.service.js]: `return ["plugAndFax", "fax", "voicefax"].indexOf(line.featureType) === -1` possible ?

**Pull request merged**